File: ios/Classes/Transfer/TrnRequestHandler.m
#import "TrnRequestHandler.h"

@implementation TrnRequestHandler

- (void)start:(UIViewController *)controller dictionary:(NSDictionary *) dictionary delegate:(id<P24TransferDelegate>) delegate {
    bool isSandbox = [[dictionary objectForKey:@"isSandbox"] boolValue];
    NSString* token = [dictionary objectForKey:@"token"];
    
    P24TrnRequestParams* params = [[P24TrnRequestParams alloc] initWithToken: token];
    params.sandbox = isSandbox;
    
    [P24 startTrnRequest:params inViewController: controller delegate: delegate];
}

@end

File: ios/Classes/Transfer/TrnRequestHandler.swift
import P24

internal struct TrnRequestHandler : TransferHandler {
    internal func start(controller: UIViewController?, dictionary: [String: Any], delegate: P24TransferDelegate) {
        let params = P24TrnRequestParams(token: (dictionary["token"] as! String))!
        params.sandbox = dictionary["isSandbox"] as! Bool
        P24.startTrnRequest(params, in: controller, delegate: delegate)
    }
}
